src/policy/rbf.cpp

```diff
@@ -181,3 +181,54 @@ std::optional<std::string> PaysForRBF(CAmount original_fees,
     }
     return std::nullopt;
 }
+
+std::optional<std::string> CheckConflictTopology(const CTxMemPool::setEntries& direct_conflicts)
+{
+    for (const auto& direct_conflict : direct_conflicts) {
+        const auto desc_count = direct_conflict->GetCountWithDescendants() - 1;
+        if (desc_count == 0) {
+            // It should be by itself, or be a child in a parent-child cluster
+            if (direct_conflict->GetCountWithAncestors() > 2) {
+                return strprintf("Child transaction has too many ancestors");
+            }
+            if (direct_conflict->GetCountWithAncestors() == 2) {
+                const auto& parent = direct_conflict->GetMemPoolParentsConst().begin();
+                if (parent->get().GetCountWithDescendants() > 1) {
```

Comment:
Should this be "> 2"?

Reply:
I have a failing unit test for this :facepalm: